src/test/javascript/spec/component/exam/manage/exercise-groups.component.spec.ts

```diff
@@ -0,0 +1,256 @@
+    const course = new Course();
+    course.id = 456;
+
+    const exam = new Exam();
+    exam.course = course;
+    exam.id = 123;
+
+    let groups: ExerciseGroup[];
+
+    let comp: ExerciseGroupsComponent;
+    let fixture: ComponentFixture<ExerciseGroupsComponent>;
+
+    let exerciseGroupService: ExerciseGroupService;
+    let examManagementService: ExamManagementService;
+    let jhiEventManager: JhiEventManager;
+    let modalService: NgbModal;
+    let router: Router;
+
+    const data = of({ exam });
+    const route = ({ snapshot: { paramMap: convertToParamMap({ courseId: course.id, examId: exam.id }) }, data } as any) as ActivatedRoute;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule, TranslateTestingModule, RouterModule],
+            declarations: [
+                ExerciseGroupsComponent,
+                ExamExerciseRowButtonsComponent,
+                ProgrammingExerciseInstructorStatusComponent,
+                MockComponent(AlertComponent),
+                MockComponent(AlertErrorComponent),
+                MockComponent(BuildPlanLinkDirective),
+                MockDirective(DeleteButtonDirective),
+                MockDirective(HasAnyAuthorityDirective),
+                MockDirective(NgbTooltip),
+                MockPipe(TranslatePipe),
+            ],
+            providers: [
+                { provide: ActivatedRoute, useValue: route },
+                { provide: Router, useClass: MockRouter },
+                { provide: NgbModal, useClass: MockNgbModalService },
+            ],
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(ExerciseGroupsComponent);
+        comp = fixture.componentInstance;
+
+        exerciseGroupService = TestBed.inject(ExerciseGroupService);
+        examManagementService = TestBed.inject(ExamManagementService);
+        jhiEventManager = TestBed.inject(JhiEventManager);
+        modalService = TestBed.inject(NgbModal);
+        router = TestBed.inject(Router);
+
+        groups = [
+            {
+                id: 0,
+                exercises: [
+                    { id: 3, type: ExerciseType.TEXT },
+                    { id: 4, type: ExerciseType.PROGRAMMING },
+                ],
+            } as ExerciseGroup,
+            { id: 1 } as ExerciseGroup,
+            { id: 2 } as ExerciseGroup,
+        ];
+    });
+
+    // Always initialized and bind before tests
+    beforeEach(fakeAsync(() => {
+        fixture.detectChanges();
+        tick();
+    }));
+
+    it('loads the exercise groups', fakeAsync(() => {
+        const mockResponse = new HttpResponse<Exam>({ body: exam });
+
+        spyOn(examManagementService, 'find').and.returnValue(of(mockResponse));
+
+        comp.loadExerciseGroups().subscribe((response) => {
+            expect(response.body).not.toBeNull();
+            expect(response.body!.id).toEqual(exam.id);
+        });
+
+        tick();
```

Comment:
What's the tick for since you're not asserting anything afterwards?

Reply:
That's because I'm testing `loadExerciseGroups()` which returns an `Observable`, so I have to `expect` in the `subscribe` handler. The `tick()` just triggers `subscribe` and executes the assertions 🙂